macaroonbakery/codec.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2017 Canonical Ltd.
+# Licensed under the LGPLv3, see LICENCE file for details.
+
+import base64
+import json
+
+from nacl.public import Box, PublicKey
+from nacl.encoding import Base64Encoder
+import six
+
+import bakery
+import macaroon
+
+
+def encode_caveat(condition, root_key, third_party_info, key, ns):
+    '''Encrypt a third-party caveat.
+
+    The third_party_info key holds information about the
+    third party we're encrypting the caveat for; the key is the
+    public/private key pair of the party that's adding the caveat.
+
+    The caveat will be encoded according to the version information
+    found in third_party_info.
+
+    @param condition string
+    @param root_key bytes
+    @param third_party_info object
+    @param key nacl key
+    @param ns not used yet
+    @return a base64 encoded string
+    '''
+    if third_party_info.version == bakery.BAKERY_V1:
+        return _encode_caveat_v1(condition, root_key,
+                                 third_party_info.public_key, key)
+
+    raise NotImplementedError('only bakery v1 supported')
+
+
+def _encode_caveat_v1(condition, root_key, third_party_pub_key, key):
+    '''Create a JSON-encoded third-party caveat.
+
+    The third_party_pub_key key represents the public key of the third party
+    we're encrypting the caveat for; the key is the public/private key pair of
+    the party that's adding the caveat.
+
+    @param condition string
+    @param root_key bytes
+    @param third_party_pub_key nacl public key
+    @param key nacl private key
+    @return a base64 encoded string
+    '''
+    plain_data = json.dumps(
```

Comment:
Is this indentation style preferred over:

    plain_data = json.dumps({
        'RootKey': base64.b64encode(root_key).decode('ascii'),
        'Condition': condition,
    })

?

(that's the style you are using below, too)